### PR TITLE
[CMake] Enable clang-format command on Windows

### PR DESCRIPTION
**Pull request by lukaszstolarczuk**
It seems like list of args for Windows was too long. Split cppformat commands into two, each formatting half of passed files. Re-enable 'check-generated' check on Windows CI again.

ref. #888

### CODE REVIEW

File: scripts/templates/stype_map_helpers.hpp.mako [hunk 1]
```diff
@@ -7,7 +8,7 @@ from templates import helper as th
     x=tags['$x']
     X=x.upper()
 %>
-// This file is autogenerated from the template at ${self.template.filename}
+// This file is autogenerated from the template at ${os.path.dirname(self.template.filename)}/${os.path.basename(self.template.filename)}
```
Comment (pbalcer) [inline scripts/templates/stype_map_helpers.hpp.mako:11]: is this related?

**Review by pbalcer — APPROVED** [review]

File: scripts/templates/stype_map_helpers.hpp.mako (hunk 1 above)
Comment (lukaszstolarczuk) [reply scripts/templates/stype_map_helpers.hpp.mako:11]: yeah, generated file contains a comment:
```
// This file is autogenerated from the template at templates/stype_map_helpers.hpp.mako
```

on Windows, there's different separator (-> `\`) and it's displayed as a diff in the generated files (causing failure)

Comment (lukaszstolarczuk) [reply scripts/templates/stype_map_helpers.hpp.mako:11]: alternatively we could just drop this comment 😉

Comment (pbalcer) [reply scripts/templates/stype_map_helpers.hpp.mako:11]: ok, lets leave it. thanks.

**Comment by lukaszstolarczuk** [comment]
Rebased with new `source checks` workflow.

**Comment by lukaszstolarczuk** [comment]
ref. #2158

File: cmake/helpers.cmake [hunk 2]
```diff
@@ -28,16 +28,22 @@ function(add_cppformat name)
...
+        add_custom_target(cppformat-${name}-1
+            COMMAND ${CLANG_FORMAT} --style=file --i ${selected_files}
+            COMMENT "Format CXX source files (pt 1)"
+        )
+
+        list(SUBLIST ARGN 251 -1 selected_files)
+        add_custom_target(cppformat-${name}-2
+            COMMAND ${CLANG_FORMAT} --style=file --i ${selected_files}
+            COMMENT "Format CXX source files (pt 2)"
+        )
```
Comment (kbenzie) [inline cmake/helpers.cmake:42]: A custom target can contain multiple commands, there's no need for multiple `cppformat-${name}` targets. e.g.

```cmake
add_custom_target(example
  COMMAND echo first
  COMMAND echo second
  COMMENT Multiple commands example
)
```

Has this output when invoking the `example` target:

```
first
second
```

```suggestion
        # Split args into 2 parts (in Windows the list is probably too long)
        list(SUBLIST ARGN 0 250 selected_files_1)
        list(SUBLIST ARGN 251 -1 selected_files_2)
        add_custom_target(cppformat-${name}
            COMMAND ${CLANG_FORMAT} --style=file --i ${selected_files_1}
            COMMAND ${CLANG_FORMAT} --style=file --i ${selected_files_2}
            COMMENT "Format CXX source files (pt 1)"
        )
```

Comment (lukaszstolarczuk) [reply cmake/helpers.cmake]: thx! that's super nice, applied.

**Review by kbenzie — APPROVED** [review]

**Comment by lukaszstolarczuk** [comment]
soo.. can we merge this? I'd like to set up some more changes around CI workflows - it would be nice to not have to re-base it again :)
